[9.0][IMP]stock_auto_move: auto move from push rules

zakiuu opened the pull request:
With this improvement, if the push rule is set to auto confirm, all the generated moves will be automatically processed as soon as the products are available in the source location.

As a use case to explain more the feature, In a reception on two steps, if the push rule is set to auto confirm, all the moves created in the second step will be processed automatically as soon as the first step is done.

zakmak88 committed:
[IMP]stock_auto_move: auto move from push rules

zakiuu commented:
this PR depends on https://github.com/OCA/stock-logistics-workflow/pull/358. and thanks to @Cedric-Pigeon for sharing the code

rousseldenis commented on stock_auto_move/models/stock_move.py:
```diff
@@ -0,0 +1,36 @@
+        "processed as soon as the products are available.")
+
+    @api.multi
+    def action_assign(self):
```
Missing `no_prepare=False` in args and ... in super()

rousseldenis commented on stock_auto_move/models/stock_move.py:
```diff
@@ -0,0 +1,36 @@
+        automatic_group = self.env.ref('stock_auto_move.automatic_group')
+        moves = self.filtered(
+            lambda m: m.auto_move and m.group_id != automatic_group)
+        moves.write({'group_id': automatic_group.id})
```
Functionally, why do we change the procurement group here ?
Moreover, why isn't it configurable ?

zakiuu replied:
@rousseldenis any functional improvement of the module I've to do it in a separate PR, this PR is only a migration to v9

zakiuu replied:
@rousseldenis issue fixed in #364

lmignon reviewed (approved):
@zakiuu z Your changes in https://github.com/OCA/stock-logistics-workflow/pull/359/commits/9c02939de9073e535d5ba98941d36f4ff18fe6b6 LGTM. We must wait for the merge of #358

zakmak88 committed:
[REF]stock_auto_move: remove override of _apply method

* The `_apply` method should not be overridden. In case of three step
operation with the last step set to auto move. The override of the
`_apply` method was causing the second move to be automatic as well.

zakiuu commented:
ping @lmignon @rousseldenis @Cedric-Pigeon

zakiuu commented:
@rousseldenis can you approve?

rousseldenis reviewed (approved):
LGTM. Code review

@zakiuu Maybe a test on push_apply function should be great

LoisRForgeFlow reviewed (approved):
Code review
